### PR TITLE
python39Packages.fixtures: fix build

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Drops two tests that try to monkeypatch a classmethod, which breaks due
to changes in Python 3.9. The project isn't really that well maintained
anymore and Debian started dropping those two tests in November 2020, so
let's follow suit

Also separate checkInputs from propagatedBuildInputs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
@GrahamcOfBorg build python39Packages.denonavr python39Packages.pyatmo python39Packages.requests-mock

**Comment by SuperSandro2000** [comment]
Do we want to target staging here? Could we also apply the patch on all python versions to prevent bit rot?

**Comment by mweinelt** [comment]
> Could we also apply the patch on all python versions to prevent bit rot?

`pythonAtLeast "3.9" will do that.

**Comment by rmcgibbo** [comment]
FWIW, I've been working on a new feature for my bot to try to build subsets of big PRs and this one came at the perfect time for local testing.

It looks like `pyopnsense` might be failing here with the following message from my testing
```
==================================== ERRORS ====================================
_______________ ERROR collecting pyopnsense/tests/test_client.py _______________
ImportError while importing test module '/build/pyopnsense-0.3.0/pyopnsense/tests/test_client.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/3qsj5r2w2z0dqj7ri70jifaq5ks3rc42-python3-3.9.1/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
pyopnsense/tests/test_client.py:25: in <module>
    from pyopnsense.tests import base
pyopnsense/tests/base.py:18: in <module>
    import fixtures
/nix/store/hd7dcbrzn8vxrx1mmya3yl7wh25awy00-python3.9-fixtures-3.0.0/lib/python3.9/site-packages/fixtures/__init__.py:81: in <module>
    from fixtures.fixture import (
/nix/store/hd7dcbrzn8vxrx1mmya3yl7wh25awy00-python3.9-fixtures-3.0.0/lib/python3.9/site-packages/fixtures/fixture.py:29: in <module>
    from testtools.compat import (
E   ModuleNotFoundError: No module named 'testtools'
=========================== short test summary info ============================
```

**Comment by rmcgibbo** [comment]
Perhaps the reason for that is that testtools really is supposed to be in propagatedBuildInputs?

**Comment by mweinelt** [comment]
Oh well.

> setup_requires=['pbr>0.11'],

https://github.com/testing-cabal/fixtures/blob/master/setup.py#L6

**Comment by mweinelt** [comment]
@GrahamcOfBorg build python39Packages.pyopnsense

File: pkgs/development/python-modules/fixtures/default.nix [hunk 1]
```diff
@@ -17,7 +18,23 @@ buildPythonPackage rec {
...
+    # drop tests that try to monkeypatch a classmethod, which fails on python3.9
+    # https://github.com/testing-cabal/fixtures/issues/44
+    (fetchpatch {
+       url = "https://salsa.debian.org/openstack-team/python/python-fixtures/-/raw/debian/victoria/debian/patches/remove-broken-monkey-patch-test.patch";
+       sha256 = "1s3hg2zmqc4shmnf90kscphzj5qlqpxghzw2a59p8f88zrbsj97r";
+    })
+  ];
+
+  propagatedBuildInputs = [
+    pbr
```
Comment (FRidh) [inline pkgs/development/python-modules/fixtures/default.nix:31]: it said `setup_requires`, so it should go in `nativeBuildInputs`.

Comment (mweinelt) [reply pkgs/development/python-modules/fixtures/default.nix:31]: Fixed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112854` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>hydra-unstable: log was empty</li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>home-assistant</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hydra-unstable:
<details>


`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/hydra/common.nix:133:5:

```
    |
133 |     license = licenses.gpl3;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
autoconf is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
automake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
perl is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
subversion is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `fetchFromGitHub`.
Near pkgs/development/tools/misc/hydra/common.nix:1:40:

```
  |
1 | { stdenv, nix, perlPackages, buildEnv, fetchFromGitHub
  |                                        ^
```
Unused argument: `darcs`.
Near pkgs/development/tools/misc/hydra/common.nix:3:23:

```
  |
3 | , top-git, mercurial, darcs, subversion, breezy, openssl, bzip2, libxslt
  |                       ^
```
Unused argument: `guile`.
Near pkgs/development/tools/misc/hydra/common.nix:4:3:

```
  |
4 | , guile, perl, postgresql, nukeReferences, git, boehmgc, nlohmann_json
  |   ^
```
</details>
</details>

```
reading manifest file 'src/launchpadlib.egg-info/SOURCES.txt'
writing manifest file 'src/launchpadlib.egg-info/SOURCES.txt'
running build_ext
test_credentials_save_fail_under_sudo_does_not_raise_exception (launchpadlib.tests.test_launchpad.TestCredenitialSaveFailedCallback) ... ok
test_credentials_save_failed (launchpadlib.tests.test_launchpad.TestCredenitialSaveFailedCallback) ... ok
test_default_credentials_save_failed_is_to_raise_exception (launchpadlib.tests.test_launchpad.TestCredenitialSaveFailedCallback) ... ok
test_get_token_and_login_is_deprecated (launchpadlib.tests.test_launchpad.TestDeprecatedLoginMethods) ... FAIL
test_login_is_deprecated (launchpadlib.tests.test_launchpad.TestDeprecatedLoginMethods) ... ok
test_None_launchpadlib_dir (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_anonymous_login (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith)
Test the anonymous login helper function. ... ok
test_application_name_identifies_app (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_application_name_is_propagated (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_authorization_engine_is_propagated (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_consumer_name_identifies_app (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_desktop_integration_doesnt_happen_without_consumer_name (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_dirs_created (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_dirs_created_are_changed_to_secure (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_dirs_created_are_secure (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_existing_credentials_arguments_passed_on (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_inconsistent_allow_access_levels_rejected (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith)
Catch an attempt to specify inconsistent allow_access_levels. ... ok
test_inconsistent_application_name_rejected (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith)
Catch an attempt to specify inconsistent application_names. ... ok
test_inconsistent_consumer_name_rejected (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith)
Catch an attempt to specify inconsistent application_names. ... ok
test_inconsistent_credential_save_failed (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_login_with_must_identify_application (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_max_failed_attempts_accepted (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_no_credentials_creates_new_credential (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_non_desktop_integration (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_short_service_name (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_version_is_propagated (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWith) ... ok
test_cannot_specify_both_filename_and_store (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWithCredentialsFile) ... ok
test_filename (launchpadlib.tests.test_launchpad.TestLaunchpadLoginWithCredentialsFile) ... ok
test_components_of_application_key (launchpadlib.tests.test_launchpad.TestMultipleSites) ... ok
test_same_app_different_servers (launchpadlib.tests.test_launchpad.TestMultipleSites) ... ok
test_edge_server_equivalent_string_becomes_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_edge_service_root_is_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_edge_service_root_url_becomes_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_edge_web_root_is_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_edge_web_root_url_becomes_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_edge_web_server_equivalent_string_becomes_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_lookups (launchpadlib.tests.test_launchpad.TestNameLookups)
Ensure that short service names turn into long service names. ... ok
test_short_names (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_top_level_edge_constant_becomes_production (launchpadlib.tests.test_launchpad.TestNameLookups) ... ok
test_app_must_be_identified (launchpadlib.tests.test_launchpad.TestRequestTokenAuthorizationEngine) ... ok
test_application_name_identifies_app (launchpadlib.tests.test_launchpad.TestRequestTokenAuthorizationEngine) ... ok
test_conflicting_app_identification (launchpadlib.tests.test_launchpad.TestRequestTokenAuthorizationEngine) ... ok
test_consumer_name_identifies_app (launchpadlib.tests.test_launchpad.TestRequestTokenAuthorizationEngine) ... ok
test_resource_types (launchpadlib.tests.test_launchpad.TestResourceTypeClasses) ... ok
test_service_name_with_embedded_version (launchpadlib.tests.test_launchpad.TestServiceNameWithEmbeddedVersion) ... ok
test_bad_json (launchpadlib.tests.test_http.TestAbilityToParseData)
Show that bad JSON causes an exception. ... ok
test_bad_wadl (launchpadlib.tests.test_http.TestAbilityToParseData)
Show that bad WADL causes an exception. ... ok
test_minimal_data (launchpadlib.tests.test_http.TestAbilityToParseData)
Make sure that launchpadlib can use the minimal data. ... ok
test_bad_token (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
If our token is bad, we get another one. ... ok
test_delayed_error (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
We get another token no matter when the error happens. ... ok
test_expired_token (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
If our token is expired, we get another one. ... ok
test_good_token (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
If our token is good, we never get another one. ... ok
test_many_errors (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
We'll keep getting new tokens as long as tokens are the problem. ... ok
test_other_unauthorized (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
If the token is not at fault, a 401 error raises an exception. ... ok
test_unknown_token (launchpadlib.tests.test_http.TestTokenFailureDuringRequest)
If our token is unknown, we get another one. ... ok
test_from_string (launchpadlib.tests.test_credential_store.TestAccessToken) ... ok
test_from_string_with_context (launchpadlib.tests.test_credential_store.TestAccessToken) ... ok
test_bad_unique_id_returns_none (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_corrupted_key_handled (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_keyring_returns_unicode (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_lookup_by_unique_key (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_nonencoded_key_handled (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_reused_unique_id_overwrites_old_credential (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_save_and_load (launchpadlib.tests.test_credential_store.TestKeyringCredentialStore) ... ok
test_file_only_contains_one_credential (launchpadlib.tests.test_credential_store.TestUnencryptedFileCredentialStore) ... ok
test_save_and_load (launchpadlib.tests.test_credential_store.TestUnencryptedFileCredentialStore) ... ok
test_unique_id_doesnt_matter (launchpadlib.tests.test_credential_store.TestUnencryptedFileCredentialStore) ... ok

======================================================================
FAIL: test_get_token_and_login_is_deprecated (launchpadlib.tests.test_launchpad.TestDeprecatedLoginMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/launchpadlib-1.10.13/src/launchpadlib/tests/test_launchpad.py", line 600, in test_get_token_and_login_is_deprecated
    self.assertEqual(len(caught), 1)
AssertionError: 3 != 1

----------------------------------------------------------------------
Ran 67 tests in 0.138s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=67 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=67 errors=0 failures=1>
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112854` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>hydra-unstable: log was empty</li>
  </ul>
</details>
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>home-assistant</li>
<li>python39Packages.pyopnsense</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hydra-unstable:
<details>


`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/hydra/common.nix:133:5:

```
    |
133 |     license = licenses.gpl3;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
autoconf is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
automake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
perl is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
subversion is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `fetchFromGitHub`.
Near pkgs/development/tools/misc/hydra/common.nix:1:40:

```
  |
1 | { stdenv, nix, perlPackages, buildEnv, fetchFromGitHub
  |                                        ^
```
Unused argument: `darcs`.
Near pkgs/development/tools/misc/hydra/common.nix:3:23:

```
  |
3 | , top-git, mercurial, darcs, subversion, breezy, openssl, bzip2, libxslt
  |                       ^
```
Unused argument: `guile`.
Near pkgs/development/tools/misc/hydra/common.nix:4:3:

```
  |
4 | , guile, perl, postgresql, nukeReferences, git, boehmgc, nlohmann_json
  |   ^
```
</details>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**